libraries/chain/backing_store/db_context_rocksdb.cpp

```diff
@@ -0,0 +1,749 @@
+   }
+
+   int32_t db_context_rocksdb::db_idx_long_double_find_primary(uint64_t code, uint64_t scope, uint64_t table,
+                                                               float128_t& secondary, uint64_t primary) {
+      return sec_lookup_long_double.find_primary(name{code}, name{scope}, name{table}, secondary, primary);
+   }
+
+   int32_t db_context_rocksdb::db_idx_long_double_lowerbound(uint64_t code, uint64_t scope, uint64_t table, float128_t& secondary,
+                                                             uint64_t& primary) {
+      return sec_lookup_long_double.lowerbound_secondary(name{code}, name{scope}, name{table}, secondary, primary);
+   }
+
+   int32_t db_context_rocksdb::db_idx_long_double_upperbound(uint64_t code, uint64_t scope, uint64_t table, float128_t& secondary,
+                                                             uint64_t& primary) {
+      return sec_lookup_long_double.upperbound_secondary(name{code}, name{scope}, name{table}, secondary, primary);
+   }
+
+   int32_t db_context_rocksdb::db_idx_long_double_end(uint64_t code, uint64_t scope, uint64_t table) {
+      return sec_lookup_long_double.end_secondary(name{code}, name{scope}, name{table});
+   }
+
+   int32_t db_context_rocksdb::db_idx_long_double_next(int32_t iterator, uint64_t& primary) {
+      return sec_lookup_long_double.next_secondary(iterator, primary);
+   }
+
+   int32_t db_context_rocksdb::db_idx_long_double_previous(int32_t iterator, uint64_t& primary) {
+      return sec_lookup_long_double.previous_secondary(iterator, primary);
+   }
+
+   // for primary keys in the iterator store, we don't care about secondary key
+   db_context_rocksdb::prim_key_iter_type db_context_rocksdb::primary_key_iter(int table_ei, uint64_t key, account_name payer) {
+      return prim_key_iter_type { .table_ei = table_ei, .secondary = noop_secondary, .primary = key, .payer = payer};
+   }
+
+   void db_context_rocksdb::swap(int iterator, account_name payer) {
+      primary_iter_store.swap(iterator, noop_secondary, payer);
+   }
+
+   // gets a prefix that allows for only primary key iterators
+   prefix_bundle db_context_rocksdb::get_primary_slice_in_primaries(name code, name scope, name table, uint64_t id) {
+      bytes primary_key = db_key_value_format::create_primary_key(scope, table, id);
+      return { primary_key, end_of_prefix::at_type, code };
+   }
+
+   pv_bundle db_context_rocksdb::get_primary_key_value(name code, name scope, name table, uint64_t id) {
+      prefix_bundle primary_key = get_primary_slice_in_primaries(code, scope, table, id);
+      return { primary_key, current_session.read(primary_key.full_key) };
+   }
+
+   void db_context_rocksdb::set_value(const shared_bytes& primary_key, const payer_payload& pp) {
+      current_session.write(primary_key, pp.as_payload());
+   }
+
+   void db_context_rocksdb::update_db_usage( const account_name& payer, int64_t delta, const storage_usage_trace& trace ) {
```

Comment:
This should be removed. It hides the implementation at db_context with the same impl.

Reply:
Removed